src/app/modules/myinfo/myinfo.format.ts

```diff
@@ -31,12 +33,11 @@ export const formatPhoneNumber = (
  * @returns Formatted address if minimally the `block`, `street`, `country`,and `postal` values are not empty in {@link addr}. Else return empty string.
  */
 export const formatAddress = (addr: MyInfoAddress | undefined): string => {
-  if (!addr || addr.unavailable) {
+  if (!addr || addr.source === MyInfoSource.NotApplicable || addr.unavailable) {
     return ''
   }
 
-  if (addr.type !== AddressType.Singapore) {
-    //workaround - AddressType.Unformatted should be the string "UNFORMATTED" not "Unformatted"
+  if (addr.type !== MyInfoAddressType.Singapore) {
```

Comment:
previously we used a lot of `?` safe navigation operators which may be unnecessary - do you think we could put in some logging code to determine whether the typing for MyInfoAddress can be trusted?

Reply:
addressed in https://github.com/opengovsg/FormSG/pull/1311/commits/a4a0d66dfd833c5600a9cdde14698aa7ecf35362, is this what you had in mind?

Reply:
yep, exactly! thanks